core/app/components/FormBuilder/ElementPanel/ComponentConfig/ColorPicker.tsx

```diff
@@ -0,0 +1,257 @@
+import type { ControllerRenderProps } from "react-hook-form";
+
+import { useEffect, useState } from "react";
+import { Pencil, PlusIcon, TrashIcon } from "lucide-react";
+import { useFormContext } from "react-hook-form";
+
+import type { InputComponent } from "db/public";
+import { Button } from "ui/button";
+import { Checkbox } from "ui/checkbox";
+import { ColorCircle, ColorPicker } from "ui/color";
+import { FormControl, FormDescription, FormField, FormItem, FormLabel, FormMessage } from "ui/form";
+import { Input } from "ui/input";
+import { Popover, PopoverContent, PopoverTrigger } from "ui/popover";
+import { cn } from "utils";
+
+import type { ComponentConfigFormProps, ConfigFormData } from "./types";
+
+export const FormBuilderColorPickerPopover = ({
+	color,
+	onChange,
+	presets,
+	presetsOnly,
+	label,
+	parentField,
+	idx,
+}: {
+	color: string;
+	label: string;
+	onChange: (value: { label: string; value: string }) => void;
+	presets?: { label: string; value: string }[];
+	presetsOnly?: boolean;
+	parentField: ControllerRenderProps<
+		ConfigFormData<InputComponent.colorPicker>,
+		"config.presets"
+	>;
+	idx: number;
+}) => {
+	const [isEditingLabel, setIsEditingLabel] = useState(false);
+
+	const saveLabelChange = (newLabel: string) => {
+		onChange({
+			label: newLabel,
+			value: color,
+		});
+		setIsEditingLabel(false);
+	};
+
+	return (
+		<div className="group flex w-full min-w-0 items-center rounded-md border bg-white shadow-sm hover:bg-gray-100">
+			<Popover>
+				<PopoverTrigger
+					disabled={isEditingLabel}
+					className="flex h-10 w-full min-w-0 cursor-pointer items-center gap-2 pl-3 focus:outline-none"
+					aria-label={`Select color: currently ${label || color}`}
+				>
+					<ColorCircle color={color} size="sm" className="flex-shrink-0" />
+
+					{isEditingLabel ? (
+						<Input
+							type="text"
+							className="my-0 mr-2 h-7 px-2 py-0 text-sm"
+							aria-label={`Change label for color preset ${label}`}
+							defaultValue={label || color}
+							onBlur={(e) => saveLabelChange(e.target.value)}
+							onKeyDown={(e) =>
+								e.key === "Enter" && saveLabelChange(e.currentTarget.value)
+							}
+							// the autofocus kinda nice tho
+							// eslint-disable-next-line jsx-a11y/no-autofocus
+							autoFocus
```

Comment:
apparently using `autoFocus` is "cringe" and "not very accessible" but it's really annoying to use otherwise